sys/include/usb.h

```diff
@@ -37,7 +37,7 @@ extern "C" {
 #define CONFIG_USB_PID (0x7D00)
 #else
 #error Please configure your vendor and product IDs. For development, you may \
-    set CONFIG_USB_VID=0x1209 CONFIG_USB_PID=0x7D01.
+    set USB_VID=${USB_VID_TESTING} USB_PID=${USB_PID_TESTING}.
```

Comment:
Isn't this still `CONFIG_USB_VID=${USB_VID_TESTING} CONFIG_USB_PID=${USB_PID_TESTING}`?

Reply:
When IDs are set in makefiles, currently (by rough convention in the examples, it's not like there is mechanism for that), they are set as `USB_xID` make variables, and later set as `-DCONFIG_USB_xID=${USB_xID}`.

Could certainly rename them to `CONFIG_USB_xID` in the makefile, but then both make and Kconfig would write the same variable. That's easier in some parts, but then what would the condition be on which the makefile decides on whether to set `CFLAGS += -DCONFIG_USB_xID=${CONFIG_USB_xID}`? (As I understand, Kconfig sets that automatically -- but when Kconfig is not in use, it'll still need manual exporting.)

Reply:
But it seems a bit odd to me that an error message that is present on one of our header files refers to a Makefile variable. As I see it, the message is indicating that you may set `CONFIG_USB_xID` macros to the testing values. Also, I guess we could actually hard-code the testing values here as they are not likely to change?

Reply:
The error message tries to give concrete guidance as to what to do; previously that would have been setting something in the Makefile; currently it should be "configure however you configure your things", which is harder to express concisely.

I suppose this is easier when there is a unified name `CONFIG_USB_xID` for both configuration ways, as "set CONFIG_USB_PID=7D01" can be understood both in a Makefile and a Kconfig context -- but for that I'd need a criterion based on which to decide whether a `CFLAGS += -DCONFIG_USB_PID=...` is indicated. What can that criterion be?

Reply:
If we just indicate that the macros `CONFIG_USB_xID` need to be configured, then the user can decide the way to do that, I don't know if it is up to the error message to define this.

But if we do want to give a 'concrete guidance', then this is fine, as CFLAGS is still the default configuration way for now.